src/components/AuthForm.js

```diff
@@ -363,18 +367,67 @@ export default class AuthForm extends PureComponent<
 
   constructor(props: Object) {
     super(props);
-    const { settings: { authMethods } } = this.props;
-    const defaultAuth = authMethods[0];
+    const defaultAuth = "anonymous";
     const { schema, uiSchema } = authSchemas[defaultAuth];
     this.state = {
       schema,
       uiSchema,
       formData: { authType: defaultAuth },
+      authMethods: [defaultAuth],
     };
   }
 
+  async componentDidMount() {
+    // Check which of our known auth implementations are supported by the server.
+    const { history } = this.props;
+    const knownAuthImplementations = [
+      "basicauth",
+      "account",
+      "fxa",
+      "ldap",
+      "portier",
+      "anonymous",
+    ];
+    const server = (history && history[0]) || defaultKintoServer;
+    const defaultAuth = "anonymous";
+    let supportedAuthImplementations = [defaultAuth];
+    setupClient({ authType: "anonymous", server: server });
+    const client = getClient();
+    try {
+      const info = await client.fetchServerInfo();
+      console.log("fetchServerInfo", info.capabilities);
+      knownAuthImplementations.forEach(authImplementation => {
+        if (authImplementation in info.capabilities) {
+          supportedAuthImplementations.push(authImplementation);
+        }
+      });
+      console.log(
+        "supported auth implementations",
+        supportedAuthImplementations
+      );
+      this.setState({
+        ...this.state,
+        authMethods: supportedAuthImplementations,
+      });
+    } catch (error) {
+      console.log("error!!!", error);
+    }
+  }
+
   onChange = ({ formData }: { formData: Object }) => {
-    const { authType } = formData;
+    const { authType, server } = formData;
+    if (this.state.formData.server !== server) {
+      // Server changed, request its capabilities and check what auth methods
+      // it supports.
+      setupClient({ authType: "anonymous", server: server });
+      const client = getClient();
+      client
+        .fetchServerInfo()
+        .then(info => {
+          console.log("fetchServerInfo", info);
+        })
+        .catch(() => console.log("error"));
```

Comment:
nit: use `await` (and possibly reuse code with `componentDidMount`

Reply:
yeah, as explained above, this code is only a placeholder (a previous version of the code i didn't mean to commit...) I'll finalize this tomorrow ;)